src/base/NekRSSeparateDomainProblem.C

```diff
@@ -224,6 +253,40 @@ NekRSSeparateDomainProblem::sendBoundaryTemperatureToNek()
     temperature(e, *_toNekRS_temp);
 }
 
+void
+NekRSSeparateDomainProblem::sendBoundaryScalarToNek(const int scalarId)
+{
+  auto & solution = _aux->solution();
+  auto sys_number = _aux->number();
+
+  if (_first)
+  {
+    _serialized_solution->init(_aux->sys().n_dofs(), false, SERIAL);
+    _first = false;
+  }
+
+  solution.localize(*_serialized_solution);
+
+  auto & mesh = _nek_mesh->getMesh();
+
+  // check which scalar PP to pass to NekRS
+  const PostprocessorValue * scalarValue = nullptr;
+
+  if(scalarId==1)
```

Comment:
With code like this, we run the risk of future updates that add scalars, but we forget to extend this part. What I'd do instead here is change the `sendBoundaryScalarToNek` function to take the `scalarValue` as an input, then call 

```
sendBoundaryScalarToNek(1, _toNekRS_scalar01);
```

Reply:
Yes good point, I'll change this